model/job/trigger_cron.go

```diff
@@ -46,11 +46,32 @@ func NewEveryTrigger(infos *TriggerInfos) (*CronTrigger, error) {
 	}, nil
 }
 
+// NewMonthlyTrigger returns a new instance of CronTrigger given the specified
+// options as @monthly. It will take a random day/hour in the possible range to
+// spread the triggers from the same app manifest.
+func NewMonthlyTrigger(infos *TriggerInfos) (*CronTrigger, error) {
+	spec, err := periodicParser.Parse(MonthlyKind, infos.Arguments)
+	if err != nil {
+		return nil, ErrMalformedTrigger
+	}
+	seed := infos.Domain + "/" + infos.WorkerType
+	crontab := spec.ToRandomCrontab(seed)
+	schedule, err := cronParser.Parse(crontab)
+	if err != nil {
+		return nil, ErrMalformedTrigger
+	}
+	return &CronTrigger{
+		TriggerInfos: infos,
+		sched:        schedule,
+		done:         make(chan struct{}),
+	}, nil
+}
+
```

Comment:
I guess a common function could be used for  `NewMonthlyTrigger`,  `NewWeeklyTrigger` and `NewDailyTrigger` ?